ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -419,13 +420,47 @@ impl<'a> Assemble<'a> {
 
         match inst.binop() {
             BinOp::Add => {
-                let size = lhs.byte_size(self.m);
+                let byte_size = lhs.byte_size(self.m);
+                match (&lhs, &rhs) {
+                    (Operand::Const(cidx), Operand::Var(_))
+                    | (Operand::Var(_), Operand::Const(cidx)) => {
+                        // Addition involves a constant. We may be able to emit more optimal code.
+                        let Const::Int(ctyidx, v) = self.m.const_(*cidx) else {
+                            unreachable!()
+                        };
+                        let Ty::Integer(bit_size) = self.m.type_(*ctyidx) else {
+                            unreachable!()
+                        };
+                        // If it's a 64-bit add and the numeric value of the constant can be
+                        // expressed in 32-bits...
+                        //
+                        // We are only optimising (exactly) 64-bit add operations for now, but
+                        // there is certainly oppertunity to optimised other cases later.
+                        //
+                        // We could have used Rust `as` casts to truncate and sign-extend here,
+                        // since we are currently only dealing with i32s and i64s, but if/when we
+                        // want to cover operations on other "odd-bit-size" integers we will need
+                        // these custom implementations.
+                        if *bit_size == 64 && v.truncate(32).sign_extend(32, 64) == *v {
```

Comment:
We could, of course, easily handle `i32` here too!

Reply:
Yes, in theory we could handle many different cases using this one instruction encoding, but I wanted to do the MVP in this first PR.

Reply:
Ah, I  just realised that you mean we could optimise using `add r32, imm32`?

Yep, as another PR if that's ok?